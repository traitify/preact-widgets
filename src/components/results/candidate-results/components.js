export CandidateResults from "./index";
export PersonalityArchetype from "./personality-archetype";
export PersonalityDetails from "./personality-details";
export PersonalityDimensionColumn from "./personality-dimension-column";
export PersonalityDimensionColumns from "./personality-dimension-columns";
export PersonalityDimensionDetails from "./personality-dimension-details";
export PersonalityDimensions from "./personality-dimensions";
