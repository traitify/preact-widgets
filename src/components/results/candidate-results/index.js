import {
  faThLarge,
  faUser
} from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import Icon from "lib/helpers/icon";
import withTraitify from "lib/with-traitify";
import Guide from "components/results/guide";
import PersonalityArchetype from "components/results/personality/archetype/heading";
import PersonalityDimensions from "components/results/personality/dimension/list";
import PersonalityTips from "components/results/personality/archetype/tips";
import PersonalityTraits from "components/results/personality/trait/list";
import style from "../style.scss";

function CandidateResults(props) {
  const {getOption, translate} = props;
  let disabledComponents = getOption("disabledComponents") || [];
  const allowHeaders = getOption("allowHeaders");
  const thirdPerson = getOption("perspective") === "thirdPerson";
  if(!thirdPerson) { disabledComponents = [...disabledComponents, "PersonalityPitfalls"]; }

  return (
    <section>
      {allowHeaders && (
        <p className={style.heading}><Icon icon={faUser} /> {translate("personality_type")}</p>
      )}
      <PersonalityArchetype {...props} />
      {allowHeaders && (
        <p className={style.heading}><Icon icon={faThLarge} /> {translate("personality_details")}</p>
      )}
      <PersonalityTips {...props} />
      {allowHeaders && (
        <p className={style.lessMarginBottom}><Icon icon={faThLarge} /> {translate("personality_advice")}</p>
      )}
      <PersonalityDimensions {...props} disabledComponents={disabledComponents} />
      {thirdPerson && (
        <div>
          {allowHeaders && (
            <p className={style.heading}><Icon icon={faThLarge} /> {translate("interview_guide_heading")}</p>
          )}
          <Guide {...props} />
        </div>
      )}
      {allowHeaders && (
        <p className={style.paddingTop}><Icon icon={faThLarge} /> {translate("personality_traits")}</p>
      )}
      <PersonalityTraits {...props} />
    </section>
  );
}

CandidateResults.propTypes = {
  getOption: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired
};

export default withTraitify(CandidateResults);
