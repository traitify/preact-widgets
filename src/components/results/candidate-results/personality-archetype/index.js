import PropTypes from "prop-types";
import {Component} from "react";
import DangerousHTML from "lib/helpers/dangerous-html";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

class PersonalityArchetype extends Component {
  static defaultProps = {assessment: null, deck: null}
  static propTypes = {
    assessment: PropTypes.shape({archetype: PropTypes.object}),
    deck: PropTypes.shape({name: PropTypes.string.isRequired}),
    followDeck: PropTypes.func.isRequired,
    getOption: PropTypes.func.isRequired,
    isReady: PropTypes.func.isRequired,
    options: PropTypes.shape({archetype: PropTypes.object}).isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityArchetype.initialized", this);
    this.props.followDeck();
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityArchetype.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }
    if(!this.props.isReady("deck")) { return null; }

    const disabledComponents = this.props.getOption("disabledComponents") || [];
    if(disabledComponents.includes("PersonalityArchetype")) { return null; }

    const personality = this.props.assessment.archetype;
    if(!personality) { return null; }
    const badge = personality.details.find(({title}) => title === "Badge");
    const perspective = this.props.getOption("perspective");
    const description = (
      perspective === "thirdPerson"
        ? personality.details.find(({title}) => title === "Hiring Manager Description")
        : personality.details.find(({title}) => title === "Candidate Description")
    );
    const splitIfPresent = description && style.badgeAndName;

    return (
      <div className={style.container}>
        {perspective === "thirdPerson" ? (
          <div className={style.thirdPersonDetails}>
            <div className={splitIfPresent}>
              {badge && <img alt={personality.name} src={badge.body} />}
              <DangerousHTML
                html={this.props.translate("personality_heading_third_person", {
                  deck_name: this.props.deck.name,
                  personality: `<br /><span>${personality.name}</span><br />`
                })}
                className={style.personalityHeading}
                tag="h2"
              />
            </div>
            {description && <span className={style.divider} />}
            {description && <span className={style.body}>{description.body}</span>}
          </div>
        ) : [
          <div key="heading" className={style.details}>
            {badge && <img alt={personality.name} src={badge.body} />}
            <DangerousHTML
              html={this.props.translate("personality_heading", {
                deck_name: this.props.deck.name,
                personality: `<span>${personality.name}</span>`
              })}
              tag="h2"
            />
            {description && <p>{description.body}</p>}
          </div>,
          <div key="meaning" className={style.meaning}>
            <DangerousHTML html={this.props.translate("candidate_description_for_archetype_html")} tag="p" />
          </div>
        ]}
      </div>
    );
  }
}

export {PersonalityArchetype as Component};
export default withTraitify(PersonalityArchetype);
