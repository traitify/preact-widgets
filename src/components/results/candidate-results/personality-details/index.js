import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style";

class PersonalityDetails extends Component {
  static defaultProps = {assessment: null}
  static propTypes = {
    assessment: PropTypes.shape({archetype: PropTypes.object}),
    getOption: PropTypes.func.isRequired,
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDetails.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDetails.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }

    const personality = this.props.assessment.archetype || {};
    const {details} = personality;
    if(!details) { return null; }

    const disabledComponents = this.props.getOption("disabledComponents") || [];
    let disableSettings = disabledComponents.includes("PersonalitySettings");
    let disableTools = disabledComponents.includes("PersonalityTools");
    const settings = details.filter(({title}) => (title === "Settings that Work for You")).map(({body}) => body);
    const tools = details.filter(({title}) => (title === "Tools to Use")).map(({body}) => body);
    if(settings.length === 0) { disableSettings = true; }
    if(tools.length === 0) { disableTools = true; }
    if(disableSettings && disableTools) { return null; }

    return (
      <div className={style.container}>
        {!disableTools && (
          <div className={style.detail}>
            <div className={style.content}>
              <div className={style.bar} style={{background: "#008dc7"}} />
              <h4 className={style.title} style={{color: "#008dc7"}}>{this.props.translate("candidate_heading_for_tools")}</h4>
              <ul className={style.description}>
                {tools.map((tool) => (
                  <li key={tool}>{tool}</li>
                ))}
              </ul>
            </div>
          </div>
        )}
        {!disableSettings && (
          <div className={style.detail}>
            <div className={style.content}>
              <div className={style.bar} style={{background: "#32be4b"}} />
              <h4 className={style.title} style={{color: "#32be4b"}}>{this.props.translate("candidate_heading_for_settings")}</h4>
              <ul className={style.description}>
                {settings.map((setting) => (
                  <li key={setting}>{setting}</li>
                ))}
              </ul>
            </div>
          </div>
        )}
        <div className={style.clearfix} />
      </div>
    );
  }
}

export {PersonalityDetails as Component};
export default withTraitify(PersonalityDetails);
