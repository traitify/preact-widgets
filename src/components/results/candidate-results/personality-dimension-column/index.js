import PropTypes from "prop-types";
import {Component} from "react";
import {rgba} from "lib/helpers/color";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

class PersonalityDimensionColumn extends Component {
  static propTypes = {
    translate: PropTypes.func.isRequired,
    type: PropTypes.shape({
      personality_type: PropTypes.shape({
        badge: PropTypes.shape({
          color_1: PropTypes.string.isRequired,
          image_medium: PropTypes.string.isRequired
        }).isRequired,
        level: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      }).isRequired,
      score: PropTypes.number.isRequired
    }).isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDimensionColumn.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDimensionColumn.updated", this);
  }
  render() {
    const {translate, type: {personality_type: {badge, level, name}, score}} = this.props;
    const color = `#${badge.color_1}`;
    let backgroundClass;
    switch(true) {
      case score <= 3: backgroundClass = "low"; break;
      case score <= 6: backgroundClass = "medium"; break;
      default: backgroundClass = "high";
    }

    return (
      <li className={style.container}>
        <div>
          <div className={style.icon} style={{border: `3px solid ${color}`, background: rgba(color, 8.5)}}>
            <img src={badge.image_medium} alt={`${name} ${translate("badge")}`} />
          </div>
          <h3>{name}</h3>
          <h2>{level}</h2>
        </div>
        <div className={`${style.background} ${style[backgroundClass]}`} style={{background: rgba(color, 10)}} />
      </li>
    );
  }
}

export {PersonalityDimensionColumn as Component};
export default withTraitify(PersonalityDimensionColumn);
