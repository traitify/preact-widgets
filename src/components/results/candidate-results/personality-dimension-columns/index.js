import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import PersonalityDimensionColumn from "../personality-dimension-column";
import style from "./style";

class PersonalityDimensionColumns extends Component {
  static defaultProps = {assessment: null}
  static propTypes = {
    assessment: PropTypes.shape({personality_types: PropTypes.array}),
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDimensionColumns.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDimensionColumns.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }

    const types = this.props.assessment.personality_types.sort((x, y) => {
      const xDetail = x.personality_type.details.find(({title}) => title === "Position") || {};
      const yDetail = y.personality_type.details.find(({title}) => title === "Position") || {};

      return (xDetail.body || 1) - (yDetail.body || 1);
    });

    return (
      <div className={style.container}>
        <p>{this.props.translate("candidate_description_for_dimensions")}</p>
        <ul className={style.columns}>
          {types.map((type) => (
            <PersonalityDimensionColumn
              key={type.personality_type.id}
              type={type}
              {...this.props}
            />
          ))}
        </ul>
        <div className={style.columnsBuffer} />
      </div>
    );
  }
}

export {PersonalityDimensionColumns as Component};
export default withTraitify(PersonalityDimensionColumns);
