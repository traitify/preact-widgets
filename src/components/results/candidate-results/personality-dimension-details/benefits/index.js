import PropTypes from "prop-types";
import {rgba} from "lib/helpers/color";
import style from "../style";

export default function Benefits(props) {
  const {badge, benefits, header} = props;
  const color = `#${badge.color_1}`;

  return (
    <div>
      <h3>{header}</h3>
      <ul className={style.detailsGroup}>
        {benefits.map((benefit) => (
          <li key={benefit} style={{background: rgba(color, 50)}}>{benefit}</li>
        ))}
      </ul>
    </div>
  );
}

Benefits.propTypes = {
  badge: PropTypes.shape({
    color_1: PropTypes.string.isRequired
  }).isRequired,
  benefits: PropTypes.arrayOf(PropTypes.object).isRequired,
  header: PropTypes.string.isRequired
};
