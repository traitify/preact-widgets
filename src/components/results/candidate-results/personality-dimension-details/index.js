import PropTypes from "prop-types";
import {Component} from "react";
import {detailWithPerspective} from "lib/helpers";
import {rgba} from "lib/helpers/color";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style";

class PersonalityDimensionDetails extends Component {
  static propTypes = {
    getOption: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    type: PropTypes.shape({
      personality_type: PropTypes.shape({
        badge: PropTypes.shape({
          color_1: PropTypes.string.isRequired,
          image_medium: PropTypes.string.isRequired
        }).isRequired,
        details: PropTypes.array.isRequired,
        level: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      }).isRequired
    }).isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDimensionDetails.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDimensionDetails.updated", this);
  }
  render() {
    const {translate, type: {personality_type: {badge, details, level, name}}} = this.props;
    const color = `#${badge.color_1}`;
    const benefits = details.filter(({title}) => (title === "Benefits")).map(({body}) => body);
    const perspective = this.props.getOption("perspective") || "firstPerson";
    const options = {base: {details}, perspective};

    return (
      <li className={style.container} style={{background: rgba(color, 10), borderTop: `5px solid ${color}`, listStyle: "none"}}>
        <div className={style.side}>
          <p className={style.icon}>
            <img src={badge.image_medium} alt={`${name} ${translate("badge")}`} />
          </p>
        </div>
        <div className={style.content}>
          <h2>{name} <span style={{color}}>|</span> {level}</h2>
          {perspective === "firstPerson" && (
            <h3>{translate("candidate_heading_for_dimension", {level, name})}</h3>
          )}
          <p className={style.description}>{detailWithPerspective({...options, name: "short_description"})}</p>
        </div>
        <h3>{translate("candidate_heading_for_benefits", {level, name})}</h3>
        <ul className={style.benefits}>
          {benefits.map((benefit) => (
            <li key={benefit} style={{background: rgba(color, 50)}}>{benefit}</li>
          ))}
        </ul>
      </li>
    );
  }
}

export {PersonalityDimensionDetails as Component};
export default withTraitify(PersonalityDimensionDetails);
