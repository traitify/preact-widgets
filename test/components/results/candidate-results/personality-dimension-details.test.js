import {Component} from "components/results/candidate-results/personality-dimension-details";
import ComponentHandler from "support/component-handler";
import assessment from "support/json/assessment/dimension-based.json";

jest.mock("lib/with-traitify", () => ((value) => value));

describe("PersonalityDimensionDetails", () => {
  let props;

  beforeEach(() => {
    props = {
      translate: jest.fn().mockName("translate"),
      type: assessment.personality_types[0],
      ui: {
        current: {},
        off: jest.fn().mockName("off"),
        on: jest.fn().mockName("on"),
        trigger: jest.fn().mockName("trigger")
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      const component = new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityDimensionDetails.initialized", component.instance);
    });

    it("triggers update", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.updateProps();

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityDimensionDetails.updated", component.instance);
    });
  });

  it("renders component", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });
});
