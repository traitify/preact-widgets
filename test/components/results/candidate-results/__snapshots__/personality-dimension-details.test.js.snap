// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalityDimensionDetails renders component 1`] = `
<li
  style={
    Object {
      "background": "rgba(41,183,112,0.1)",
      "borderTop": "5px solid #29B770",
      "listStyle": "none",
    }
  }
>
  <div>
    <p>
      <img
        alt="Agreeableness undefined"
        src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
      />
    </p>
  </div>
  <div>
    <h2>
      Agreeableness
       
      <span
        style={
          Object {
            "color": "#29B770",
          }
        }
      >
        |
      </span>
       
      Medium
    </h2>
    <h3 />
    <p>
      You describe yourself as someone who has a mature and streetwise view of other people and their views. Your first instinct is likely to be to trust other people, but you do know that others can let you down. Thus, you are probably able to be direct, but are also able to temper your approach with sensitivity for what will keep relationships healthy and intact.
    </p>
  </div>
  <h3 />
  <ul>
    <li
      style={
        Object {
          "background": "rgba(41,183,112,0.5)",
        }
      }
    >
      Forgiving style
    </li>
    <li
      style={
        Object {
          "background": "rgba(41,183,112,0.5)",
        }
      }
    >
      Looks for win-win outcome
    </li>
    <li
      style={
        Object {
          "background": "rgba(41,183,112,0.5)",
        }
      }
    >
      Negotiates with sensitivity
    </li>
    <li
      style={
        Object {
          "background": "rgba(41,183,112,0.5)",
        }
      }
    >
      Realistic view of others
    </li>
    <li
      style={
        Object {
          "background": "rgba(41,183,112,0.5)",
        }
      }
    >
      Recognises imperfection 
    </li>
    <li
      style={
        Object {
          "background": "rgba(41,183,112,0.5)",
        }
      }
    >
      Relational style
    </li>
  </ul>
</li>
`;
